targets/unity-v2/Testing/Tests/Server/ServerInstanceApiTests.cs

```diff
@@ -0,0 +1,276 @@
+                AuthenticationContext = context
+            };
+
+            try
+            {
+                PlayFabServerInstanceAPI serverInstance1 = new PlayFabServerInstanceAPI(settings, context);
+                PlayFabServerInstanceAPI serverInstance2 = new PlayFabServerInstanceAPI(settings, context);
+
+                SuccessfulLoginCount = 0;
+                serverInstance1.LoginWithServerCustomId(loginRequest1, PlayFabUUnitUtils.ApiActionWrapper<ServerLoginResult>(testContext, ApiInstanceLoginSuccessCallBack), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+                serverInstance2.LoginWithServerCustomId(loginRequest2, PlayFabUUnitUtils.ApiActionWrapper<ServerLoginResult>(testContext, ApiInstanceLoginSuccessCallBack), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+            }
+            catch (Exception)
+            {
+                testContext.Fail("Multi Intance Server api can not be created");
+            }
+        }
+        private void ApiInstanceLoginSuccessCallBack(ServerLoginResult result)
+        {
+            SuccessfulLoginCount++;
+            if (SuccessfulLoginCount == 2)
+            {
+                var testContext = (UUnitTestContext)result.CustomData;
+                testContext.EndTest(UUnitFinishState.PASSED, null);
+            }
+        }
+
+        /// <summary>
+        /// SERVER API
+        /// If API settings object is not set by a customer developer for an instance of an API class then the
+        /// API instance always uses the static PlayFabSettings class (the same way as static API classes)
+        /// </summary>
+        [UUnitTest]
+        public void CheckWithNoSettings(UUnitTestContext testContext)
+        {
+            PlayFabSettings.DeveloperSecretKey = testTitleData.developerSecretKey;
+
+            //It should work with static class only
+            PlayFabServerInstanceAPI serverInstanceWithoutAnyParameter = new PlayFabServerInstanceAPI();
+            serverInstanceWithoutAnyParameter.GetAllSegments(new GetAllSegmentsRequest(), PlayFabUUnitUtils.ApiActionWrapper<GetAllSegmentsResult>(testContext, CheckWithNoSettingsSuccessCallBack), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+        }
+        private void CheckWithNoSettingsSuccessCallBack(GetAllSegmentsResult result)
+        {
+            var testContext = (UUnitTestContext)result.CustomData;
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+        /// <summary>
+        /// SERVER API
+        /// Try to check server instance with authentication context and without authentication context
+        /// </summary>
+        [UUnitTest]
+        public void CheckWithAuthContextAndWithoutAuthContext(UUnitTestContext testContext)
+        {
+            PlayFabSettings.DeveloperSecretKey = testTitleData.developerSecretKey;
```

Comment:
This should be two lines:
PlayFabSettings.TitleId = testTitleData.titleId;
PlayFabSettings.DeveloperSecretKey = testTitleData.developerSecretKey;
And both lines should be in Setup(), not in the test.